src/react/components/Badge.jsx

```diff
@@ -2,22 +2,48 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Badge as AntBadge } from 'antd';
+import { ThemeConsumer } from 'oskari-ui/util';
+import { getTextColor, getHeaderTheme } from 'oskari-ui/theme/ThemeHelper';
 import 'antd/es/badge/style/index.js';
 
-export const Badge = ({ count, inversed, ...rest }) => {
-    const style = {
-        backgroundColor: inversed ? '#333' : '#999',
-        color: '#fff',
-        fontWeight: '700',
-        whiteSpace: 'nowrap',
-        textShadow: '0 -1px 0 rgba(0,0,0,.25)'
-    };
-    return <AntBadge count={count} style={style} overflowCount={999} showZero={true} {...rest} />;
+const STYLE = {
+    fontWeight: '700',
+    whiteSpace: 'nowrap',
+    textShadow: '0 -1px 0 rgba(0,0,0,.25)'
 };
+
+export const Badge = ThemeConsumer(({
+    theme,
+    count,
+    themed,
+    inversed,
+    showZero = true,
+    overflowCount = 999,
+    color: propsColor,
+    ...rest
+}) => {
+    const useTheme = themed && theme;
+    let color = propsColor;
+    if (useTheme) {
+        color = getHeaderTheme(theme).getAccentColor(theme);
+    } else if (inversed) {
+        color = '#333';
+    } else if (!propsColor) {
+        color = '#999';
+    }
```

Comment:
Should we just list the places that badge is used, default themed to true and check if we need non-themed ones and if we can use some colors from theme of constants for inversed color etc.

Reply:
Removed inversed and themed prop. Added constant for zero. For now badge is rendered disabled like grey color if count is 0 or missing. Theme is used if color isn't given. Removed ThemedBadge.

Inversed true/false was used in layerlist. Changed it to use theme. Changes can be reverted (black badge):
`<Badge count={badgeText} />;`
=>
`<Badge color='#333' count={badgeText} />;`
in layerlist